sabnzbd/deobfuscate_filenames.py

```diff
@@ -150,65 +156,88 @@ def deobfuscate_list(filelist, usefulname):
-                continue
-            ext = get_ext(file)
-            if ext in extcounter:
-                extcounter[ext] += 1
-            else:
-                extcounter[ext] = 1
-            if extcounter[ext] >= 3 and ext not in excluded_file_exts:
-                # collection, and extension not yet in excluded_file_exts, so add it
-                excluded_file_exts = (*excluded_file_exts, ext)
-                logging.debug(
-                    "Found a collection of at least %s files with extension %s, so not renaming those files",
-                    extcounter[ext],
-                    ext,
-                )
-
-        logging.debug("Trying to see if there are qualifying files to be deobfuscated")
-        # We start with he biggest file ... probably the most important file
-        filelist = sorted(filelist, key=os.path.getsize, reverse=True)
-        for filename in filelist:
-            # check that file is still there (and not renamed by the secondary renaming process below)
-            if not os.path.isfile(filename):
-                continue
-            logging.debug("Deobfuscate inspecting %s", filename)
-            # Do we need to rename this file?
-            # Criteria: big, not-excluded extension, obfuscated (in that order)
-            if (
-                os.path.getsize(filename) > MIN_FILE_SIZE
-                and get_ext(filename) not in excluded_file_exts
-                and is_probably_obfuscated(filename)  # this as last test to avoid unnecessary analysis
-            ):
-                # Rename and make sure the new filename is unique
-                path, file = os.path.split(filename)
-                # construct new_name: <path><usefulname><extension>
-                new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), get_ext(filename)))
-                logging.info("Deobfuscate renaming %s to %s", filename, new_name)
-                renamer(filename, new_name)
-                # find other files with the same basename in filelist, and rename them in the same way:
-                basedirfile, _ = os.path.splitext(filename)  # something like "/home/this/myiso"
-                for otherfile in filelist:
-                    if otherfile.startswith(basedirfile + ".") and os.path.isfile(otherfile):
-                        # yes, same basedirfile, only different extension
-                        remainingextension = otherfile.replace(basedirfile, "")  # might be long ext, like ".dut.srt"
-                        new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), remainingextension))
-                        logging.info("Deobfuscate renaming %s to %s", otherfile, new_name)
-                        # Rename and make sure the new filename is unique
-                        renamer(otherfile, new_name)
+    if par2_renaming_done:
+        # TODO really needed to quit here? We could also proceed with the other actions. Anyway:
```

Comment:
Agree with your TODO, why stop here? Maybe the par2 file only renamed 1 file out of 100.

Reply:
I tried this, but it has a lot of effects on the code, including the test code.
My proposal: I do this in a separate PR

Reply:
Agreed!